scripts/histmakers/mw_with_mu_eta_pt.py

```diff
@@ -484,12 +520,11 @@ def build_graph(df, dataset):
         return results, weightsum
 
     if not args.onlyMainHistograms:
-        syst_tools.add_QCDbkg_jetPt_hist(results, df, axes, cols, jet_pt=30)
+        syst_tools.add_QCDbkg_jetPt_hist(results, df, axes, cols, jet_pt=30, storage_type=storage_type)
 
-    if dataset.is_data:
+    if dataset.is_data or isQCDMC:
```

Comment:
Shouldn't the QCD MC still get at least the prefiring and pileup weights? In general, I understand this could be related to using the QCD MC for test of the fake rate, but there should be the possibility to keep using all weights in case one wants to use the QCD MC instead of the data-driven estimate for some data/MC comparison